cilium-health/launch/endpoint.go

```diff
@@ -102,10 +118,59 @@ func configureHealthRouting(netns, dev string, addressing *models.NodeAddressing
 	return err
 }
 
+// PingEndpoint attempts to make an API ping request to the local cilium-health
+// endpoint, and returns whether this was successful.
+func PingEndpoint() error {
+	if client == nil {
+		return fmt.Errorf("cilium-health endpoint hasn't yet been initialized")
+	}
+	_, err := client.Restapi.GetHello(nil)
+	return err
+}
+
+// CleanupEndpoint attempts to kill any existing cilium-health endpoint and
+// clean up its devices and pidfiles.
+func CleanupEndpoint(owner endpoint.Owner) {
```

Comment:
Since we create a new veth for the new endpoint the worst case scenario here is that we leak older health endpoints, right? They would never exit but also never see any traffic?

Reply:
Right. This patch is relying on the pidfile being created by the health agent, and that we can use that to determine the old instance PID and terminate it.

For the veth, it's always named the exact same so we can destroy and recreate it.

Reply:
Hmm, so how would this work out if cilium-agent crashes and the health endpoint is orphaned? I guess it's like deleting an open file? the old veth remains around for the oblivious old cilium-health and we all move on?

Reply:
I believe that if you delete one side of a veth, it cleans up both and nothing retains a reference to it. Sure, if a packet is currently being forwarded across it, it won't be cleaned up until after the packet has been transmitted, but I don't think it can end up being orphaned long-term.

The namespace for the old endpoint on the other hand would be treated like this, but it is bound to the lifetime of the process through the trap implementation in the spawn_netns script. When the process is killed, it's corresponding namespace should be cleaned up.

(Maybe we can simplify this over time, but that would probably be exceeding the scope of this bugfix)